Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -252,6 +255,28 @@ static OSStatus sampleCallback(CMBufferRef buffer, void* refcon)
     return false;
 }
 
+CMTime VideoMediaSampleRenderer::minimumUpcomingSampleTime(CMSampleBufferRef sample) const
+{
+    assertIsCurrent(dispatcher().get());
+
+    CMTime minimumSampleTime = PAL::CMSampleBufferGetPresentationTimeStamp(sample);
+
+    if (!m_hasOutOfOrderFrames)
+        return minimumSampleTime;
+
+    // The maximum queue depth possible for out of order frames with either H264 or HEVC is 16, limit looking ahead of 16 frames.
+    size_t forwardIndex = 0;
+    for (auto it = m_compressedSampleQueue.begin(); it != m_compressedSampleQueue.end() && forwardIndex < 16; ++forwardIndex, ++it) {
```

Comment:
Nit: it would be good to define a `constexpr` instead of hardcoding 16 since the same logic is in `hasIncomingOutOfOrderFrame`.